src/AppInstallerCLICore/Workflows/PortableInstallFlow.cpp

```diff
@@ -0,0 +1,451 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "PortableInstallFlow.h"
+#include "winget/Filesystem.h"
+#include "AppInstallerStrings.h"
+
+using namespace AppInstaller::Manifest;
+using namespace AppInstaller::Utility;
+using namespace AppInstaller::Registry;
+using namespace std::filesystem;
+
+namespace AppInstaller::CLI::Workflow
+{
+    namespace
+    {
+        constexpr std::wstring_view s_PathName = L"Path";
+        constexpr std::wstring_view s_PathSubkey_User = L"Environment";
+        constexpr std::wstring_view s_PathSubkey_Machine = L"SYSTEM\\CurrentControlSet\\Control\\Session Manager\\Environment";
+        constexpr std::wstring_view s_UninstallRegistryX64 = L"Software\\Microsoft\\Windows\\CurrentVersion\\Uninstall";
+        constexpr std::wstring_view s_UninstallRegistryX86 = L"Software\\Wow6432Node\\Microsoft\\Windows\\CurrentVersion\\Uninstall";
+
+        constexpr std::wstring_view s_DisplayName = L"DisplayName";
+        constexpr std::wstring_view s_DisplayVersion = L"DisplayVersion";
+        constexpr std::wstring_view s_Publisher = L"Publisher";
+        constexpr std::wstring_view s_InstallDate = L"InstallDate";
+        constexpr std::wstring_view s_URLInfoAbout = L"URLInfoAbout";
+        constexpr std::wstring_view s_HelpLink = L"HelpLink";
+        constexpr std::wstring_view s_UninstallString = L"UninstallString";
+        constexpr std::wstring_view s_WinGetInstallerType = L"WinGetInstallerType";
+        constexpr std::wstring_view s_InstallLocation = L"InstallLocation";
+        constexpr std::wstring_view s_PortableTargetFullPath = L"TargetFullPath";
+        constexpr std::wstring_view s_PortableSymlinkFullPath = L"SymlinkFullPath";
+        constexpr std::wstring_view s_PortableLinksDirectory = L"LinkDirectory";
+        constexpr std::wstring_view s_SHA256 = L"SHA256";
+
+        constexpr std::wstring_view s_WinGetPackageIdentifier = L"WinGetPackageIdentifier";
+        constexpr std::wstring_view s_WinGetSourceIdentifier = L"WinGetSourceIdentifier";
+        constexpr std::string_view s_LocalSource = "*Local"sv;
```

Comment:
Out of curiosity, is there a reason to have all of these called out explicitly, instead of creating a new `PortableAppARPEntry` struct which inherits from the base Key and contains a method that is the equivalent of `IsSamePortablePackageEntry`  ?

Reply:
Took your suggestion and created a PortableArpEntry struct